tsl/test/isolation/expected/compression_chunk_race.out

```diff
@@ -0,0 +1,131 @@
+   SELECT compression_status FROM chunk_compression_stats('sensor_data');
+   SELECT count(*) FROM sensor_data;
+ <waiting ...>
+step s2_compress: 
+   SELECT compression_status FROM chunk_compression_stats('sensor_data');
+   SELECT 1 FROM (SELECT compress_chunk(i, if_not_compressed => true) FROM show_chunks('sensor_data') i) i;
+   SELECT compression_status FROM chunk_compression_stats('sensor_data');
+   SELECT count(*) FROM sensor_data;
+   RESET client_min_messages;
+ <waiting ...>
+step s3_unlock_compression: 
+    -- Ensure that we are waiting on our debug waitpoint and one chunk
+    -- (Oid of compress_chunk_impl_start = 3379597659)
+    -- 'SELECT relation::regclass, ....' can not be used, because it returns a field with a variable length
+    SELECT locktype, mode, granted, objid FROM pg_locks WHERE not granted AND (relation::regclass is NULL or relation::regclass::text LIKE '%chunk') ORDER BY relation, locktype, mode, granted;
+    SELECT debug_waitpoint_release('compress_chunk_impl_start');
+
+locktype|mode     |granted|     objid
+--------+---------+-------+----------
+relation|ShareLock|f      |          
+advisory|ShareLock|f      |3379597659
+(2 rows)
+
+debug_waitpoint_release
+-----------------------
+                       
+(1 row)
+
+step s1_compress: <... completed>
+compression_status
+------------------
+Uncompressed      
+Uncompressed      
+(2 rows)
+
+?column?
+--------
+       1
+       1
+(2 rows)
+
+compression_status
+------------------
+Compressed        
+Compressed        
+(2 rows)
+
+  count
+-------
+1008050
+(1 row)
+
+step s2_compress: <... completed>
+compression_status
+------------------
+Uncompressed      
+Uncompressed      
+(2 rows)
+
+ERROR:  chunk "_hyper_X_X_chunk" is already compressed
+step s1_decompress: 
+   SELECT 1 FROM (SELECT decompress_chunk(i, if_compressed => true) FROM show_chunks('sensor_data') i) i;
+   SELECT compression_status FROM chunk_compression_stats('sensor_data');
+   SELECT count(*) FROM sensor_data;
+ <waiting ...>
+step s2_decompress: 
+   SELECT 1 FROM (SELECT decompress_chunk(i, if_compressed => true) FROM show_chunks('sensor_data') i) i;
+   SELECT compression_status FROM chunk_compression_stats('sensor_data');
+   SELECT count(*) FROM sensor_data;
+   RESET client_min_messages;
+ <waiting ...>
+step s3_unlock_decompression: 
+    -- Ensure that we are waiting on our debug waitpoint and one chunk
+    -- (Oid of decompress_chunk_impl_start = 2415054640)
+    -- 'SELECT relation::regclass, ....' can not be used, because it returns a field with a variable length
+    SELECT locktype, mode, granted, objid FROM pg_locks WHERE not granted AND (relation::regclass is NULL or relation::regclass::text LIKE '%chunk') ORDER BY relation, locktype, mode, granted;
+    SELECT debug_waitpoint_release('decompress_chunk_impl_start');
+
+locktype|mode         |granted|     objid
+--------+-------------+-------+----------
+relation|ExclusiveLock|f      |          
```

Comment:
Might be good to check the locktype rather than use `relation::regclass is NULL`. The relation might be null also for other lock types than advisory and relation locks.

Reply:
It is changed as discussed.